File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> v1,v2,ans;
        
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]>0){v1.push_back(nums[i]);}
            
            else{
                v2.push_back(nums[i]);
            }
        }
        for(int i=0;i<v1.size();i++)
        {
            ans.push_back(v1[i]);
            ans.push_back(v2[i]);
        }
        
        return ans;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        ans = [0] * len(nums)  # Create an initial list with the same length as nums
        negi = 1
        posi = 0
        for i in range(len(nums)):
            if nums[i] < 0:
                ans[negi] = nums[i]
                negi += 2
            else:
                ans[posi] = nums[i]
                posi += 2
        return ans
